elf_binary.cc

```diff
@@ -133,7 +132,18 @@ void ELFBinary::ReadDynSymtab(const std::map<std::string, std::string>& filename
 
         // Get version information coresspoinds to idx
         auto p = GetVersion(idx, filename_to_soname);
-        Elf_Versym v = (versym_) ? versym_[idx] : -1;
+        Elf_Versym v;
+        if (versym_) {
+            // We cannot reuse the old versym value(versym_[idx]) other than
+            // the case of is_special_ver_ndx(versym_[idx]). Because we build
+            // the new version information from scratch.
+            v = is_special_ver_ndx(versym_[idx]) ? versym_[idx] : VersionBuilder::NEED_NEW_VERNUM;
+        } else {
+            // Infer an appropriate versym value when the ELF file doesn't have
+            // version information.
+            // TODO(akawashiro) Is it correct?
+            v = ((ELF_ST_BIND(sym->st_info) == STB_GLOBAL) ? VER_NDX_GLOBAL : VER_NDX_LOCAL);
```

Comment:
I guess a better condition is `!= STB_LOCAL`, but I'm not sure at all. Do we actually have any STB_LOCAL / VER_NDX_LOCAL in any symbols except the placeholder symbol (the first entry of .dynsym)?

Reply:
OK. I changed the condition. I could not find any symbols with STB_LOCAL in shared objects of `libtorch`.

However, I find many symbols with STB_GLOBAL and VER_NDX_LOCAL. (e.g. `_ZN2at11randint_outERNS_6TensorElN3c108ArrayRefIlEENS2_8optionalINS_9GeneratorEEE` in `libtorch_python.so`). Therefore we cannot assume all symbols with STB_GLOBAL have the version information of VER_NDX_GLOBAL.

I want to merge this PR as it is, but I may fix this code in the future.